File: trunk/src/core/src/main/java/org/semachina/jena/datatype/factory/YearMonthFactory.java
package org.semachina.jena.datatype.factory;

import com.hp.hpl.jena.vocabulary.XSD;
import org.semachina.jena.datatype.types.YearMonth;
import org.semachina.jena.impl.SimpleRDFDatatypeImpl;

public class YearMonthFactory extends SimpleRDFDatatypeImpl<YearMonth> {
    public YearMonthFactory() {
        super(XSD.gYearMonth.getURI(), YearMonth.class);
    }

    @Override
    public YearMonth parseLexicalForm(String lexicalForm) throws Exception {
        return new YearMonth(lexicalForm);
    }

    @Override
    public String toLexicalForm(YearMonth cast) throws Exception {
        return cast.toString();
    }
}

File: src/core/src/main/scala/org/semachina/jena/datatype/factory/YearMonthFactory.scala
package org.semachina.jena.datatype.factory

import com.hp.hpl.jena.vocabulary.XSD
import org.semachina.jena.datatype.types.YearMonth
import org.semachina.jena.datatype.SimpleRDFDatatype


class YearMonthFactory extends SimpleRDFDatatype[YearMonth](
  XSD.gYearMonth.getURI,
  { lexicalForm: String => new YearMonth(lexicalForm) } )